docs/docs/multirole.md

```diff
@@ -0,0 +1,337 @@
+
+Multi-role functionality itself is enabled by default; no command-line parameter is required to enable it. However, there are run-time configuration options (group `enforceRole` setting and service `role` field) to control it, along with the command-line parameter `--new_group_enforce_role` to affect default behavior.
+
+While it is possible to partially use multi-role in Marathon, for maximum ease-of-use ability, it is recommended that Marathon be fully "upgraded" to multi-role. This is to say:
+
+* All top-level services (those not in groups) are moved to a group.
+* All services are updated to use the group-role
+* `enforceRole` is enabled for all top-level groups
+* The `--new_group_enforce_role` command-line parameter is set to `"top"`.
+
+# Service role field
+
+Each Marathon service now contains a "role" field. By default, this "role" field defaults to the role specified by the command line parameter `--mesos_role`, which in turn is `*` when not specified, by default. For example, if the command line parameter `--mesos_role` is `"slave_public"` (as is the default in DC/OS), and the following service is posted to Marathon `/v2/apps`:
+
+```
+{
+  "id": "/dev/bigbusiness",
+  "cmd": "sleep 3600",
+  "instances": 10,
+  "cpus": 0.05,
+  "mem": 128
+}
+```
+
+... then, the resulting Marathon service will receive the default role `"slave_public",` as follows:
+
+```
+{
+  "id": "/dev/bigbusiness",
+  "cmd": "set; sleep 3600",
+  "instances": 10,
+  "cpus": 0.05,
+  "mem": 128,
+  "role": "slave_public"
+}
+```
+
+Marathon will then proceed to revive offers for the role `"slave_public",` and match offers allocated to the role `"slave_public"` in order to launch this app.
+
+# Group role
+
+The group role is the name of the top-level group, as illustrated by the following table:
+
+| Full service id | Marathon group | Group-role |
+| --------------- | -------------- | --------- |
+| /dev/bigbusiness | /dev | "dev" |
+| /bigbusiness | / | N/A |
+| /dev/team1/bigbusiness | /dev/team1 | "dev" |
+| /frontend/ui | /frontend | "frontend" |
+
+Note that services placed directly in the root group do not have a group-role. A service may only be assigned one of two possible roles: the default role (that which is specified by `--mesos_role`), or, the group role.
+
+# Migrating existing services
+
+When an existing service role is modified from using the default Mesos role to using the group role, this is referred to as "migrating the service to multi-role".
```

Comment:
```suggestion
Migrating a service to multi-role means to modify an existing service role from using the default Mesos role to using the group role."
```

A sentence that starts with a temporal clause is harder to grok IMO – I have to read through until the explanation in order to understand what the beginning is trying to tell me …